onebusaway-android/src/main/java/org/onebusaway/android/travelbehavior/utils/TravelBehaviorUtils.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (C) 2019 University of South Florida
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.onebusaway.android.travelbehavior.utils;
+
+import com.google.android.gms.location.ActivityTransition;
+import com.google.android.gms.location.DetectedActivity;
+
+import org.onebusaway.android.BuildConfig;
+import org.onebusaway.android.app.Application;
+import org.onebusaway.android.io.elements.ObaRegion;
+import org.onebusaway.android.travelbehavior.constants.TravelBehaviorConstants;
+import org.onebusaway.android.travelbehavior.model.TravelBehaviorInfo;
+import org.onebusaway.android.util.PreferenceUtils;
+
+import android.content.Context;
+import android.os.Vibrator;
+import android.widget.Toast;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class TravelBehaviorUtils {
+
+    public static String toActivityString(int activity) {
+        switch (activity) {
```

Comment:
@cagryInside Based on https://developers.google.com/android/reference/com/google/android/gms/location/DetectedActivity, it looks like we're missing `ON_BICYCLE` and `TILTING` - could you please add those too?

Reply:
Activity transition API only supports 
   * `IN_VEHICLE`
   * `ON_BICYCLE`
   * `RUNNING`
   * `STILL`
   * `WALKING`
activities as shown in https://developer.android.com/guide/topics/location/transitions.html#register.

That's why I only registered  these events in https://github.com/OneBusAway/onebusaway-android/pull/988/files#diff-8dc92f32bc698f47c67820a05841eca9R276. 

And, here you are right, I forget to add `ON_BICYCLE` event.

Reply:
The more I think about this, we should probably directly capture the `DetectedActivity.getType()` integer as well.  Right now, if Google adds more detected activity types via a Google Play Services update they would all show up as "UNKNOWN" in our database until we update OBA Android.  If we capture the integer value directly from `DetectedActivity.getType()` we'll automatically capture new activity types as they are added.  We can keep the string value as well to make it easier to read.  We should probably only set the string value to `UNKNOWN` for `UNKNOWN = 4` integer value, though - if we get an unknown integer value, we should probably just leave the field null until we can post-process it and add a string description later.

Reply:
We only register `IN_VEHICLE, ON_BICYCLE, WALKING, STILL, and RUNNING` activities. This design choose was made because the Transition API only supports these activities.  If we register other activities such as `TILTING`, `ActivityRecognition`'s `requestActivityTransitionUpdates` method throws an exception saying `ActivityTransitionRequest specified an unsupported transition activity type.`